cppcache/integration/test/PartitionRegionOpsTest.cpp

```diff
@@ -168,97 +186,67 @@ void putPartitionedRegionWithRedundancyServerGoesDown(bool singleHop) {
 
   int ENTRIES = 30;
 
-  putEntries(region, ENTRIES, 0);
-
-  cluster.getServers()[1].stop();
-
-  putEntries(region, ENTRIES, 1);
-
-  cluster.getServers()[1].start();
-
-  putEntries(region, ENTRIES, 2);
-}
+  putEntries(region, ENTRIES, 0, isAllOp);
 
-void getPartitionedRegionWithRedundancyServerGoesDown(bool singleHop) {
-  Cluster cluster{LocatorCount{1}, ServerCount{2}};
-  cluster.start();
-  cluster.getGfsh()
-      .create()
-      .region()
-      .withName("region")
-      .withType("PARTITION")
-      .withRedundantCopies("1")
-      .execute();
-
-  auto cache = createCache();
-  auto pool = createPool(cluster, cache, singleHop);
-  auto region = setupRegion(cache, pool);
-
-  int ENTRIES = 30;
-
-  putEntries(region, ENTRIES, 0);
-
-  getEntries(region, ENTRIES);
+  getEntries(region, ENTRIES, isAllOp);
 
   cluster.getServers()[1].stop();
 
-  getEntries(region, ENTRIES);
+  getEntries(region, ENTRIES, isAllOp);
 
   cluster.getServers()[1].start();
 
-  getEntries(region, ENTRIES);
+  getEntries(region, ENTRIES, isAllOp);
 }
 
 /**
  * In this test case we verify that in a partition region with redundancy
- * when one server goes down, all gets are still served.
+ * when one server goes down, all puts and gets are still served.
  * Single-hop is enabled in the client.
  * It can be observed in the logs that when one of the server goes down
  * the bucketServerLocations for that server are removed from the
  * client metadata.
  */
 TEST(PartitionRegionOpsTest,
-     getPartitionedRegionWithRedundancyServerGoesDownSingleHop) {
+     putgetPartitionedRegionWithRedundancyServerGoesDownSingleHop) {
```

Comment:
Good call. Since I didn't write this test (or review it), I didn't want to trash it too badly. I think we can do away with many of the comments, which are mostly just duplicated for the four tests, in favor of using better test names. How do you feel about just breaking the single function with parameters (putgetPartitionedRegionWithRedundancyServerGoesDown(singleHop, useAll)) into the four separate aptly named functions with no parameters:

    putgetHA
    putgetHAWithSingleHop
    putAllgetAllHA
    putAllGetAllHASingleHop

I think the HA implies failover so we could nuke the comments.

Reply:
An appropriately encompassing test class name with appropriately descriptive test names should cover it. The problem with the old style was it preferred keyboard conservation over readability and maintainability. I would rather see duplicate code with one minor change than a method with several boolean flags.

Reply:
I've refactored to use better test names and less parameters in favor of simpler functions. Decided to leave in most of the test comments as they help describe the complex flow of operations including stop and restarting servers.